packages/core/src/oidc/grants/token-exchange/index.ts

```diff
@@ -76,11 +76,16 @@ export const buildHandler: (
     scopes: oidcScopes,
   } = providerInstance.configuration();
 
-  const { userId, subjectTokenId } = await validateSubjectToken(
+  const { userId, subjectTokenId } = await validateSubjectToken({
     queries,
-    String(params.subject_token),
-    String(params.subject_token_type)
-  );
+    subjectToken: String(params.subject_token),
+    subjectTokenType: String(params.subject_token_type),
+    AccessToken,
+    jwtVerificationOptions: {
+      localJWKSet: envSet.oidc.localJWKSet,
+      issuer: envSet.oidc.issuer,
+    },
+  });
```

Comment:
How about making this field optional, and loading the envSet variable directly at the function level?

Reply:
This is not global env set, it is tenant-level `envSet`, not `EnvSet`.